File: python/selection_sort.py
def selection_sort(arr):
    n = len(arr)
    for i in range(n):
        min_idx = i
        print(f"Iterasi {i+1}: {arr}")
        for j in range(i+1, n):
            if arr[j] < arr[min_idx]:
                min_idx = j
        arr[i], arr[min_idx] = arr[min_idx], arr[i]
    return arr

n = int(input("Jumlah data: "))
data = list(map(int, input("Masukkan data: ").split()))

print("Hasil:", selection_sort(data))

File: cpp/selection_sort.cpp
#include <iostream>
using namespace std;

void print(int arr[], int n) {
    for(int i=0;i<n;i++) cout << arr[i] << " ";
    cout << endl;
}

void selectionSort(int arr[], int n) {
    for(int i=0;i<n;i++) {
        cout << "Iterasi " << i+1 << ": ";
        print(arr, n);

        int min_idx = i;
        for(int j=i+1;j<n;j++) {
            if(arr[j] < arr[min_idx]) {
                min_idx = j;
            }
        }
        swap(arr[i], arr[min_idx]);
    }
}

int main() {
    int n;
    cout << "Jumlah data: ";
    cin >> n;

    int arr[n];
    cout << "Masukkan data: ";
    for(int i=0;i<n;i++) cin >> arr[i];

    selectionSort(arr, n);

    cout << "Hasil sorting: ";
    print(arr, n);
}
